retro-lcd/usr/share/asteroid-launcher/watchfaces/retro-lcd.qml

```diff
@@ -13,6 +14,73 @@ Item {
             verticalCenter: parent.verticalCenter
         }
     }
+    Canvas {
+        id: digitalambient
+        z: 0
+
+        anchors.fill: parent
+        contextType: "2d"
+        antialiasing: true
+        smooth: true
+        renderStrategy: Canvas.Threaded
+        visible: displayAmbient
+        
+        onPaint: {
+            var ctx = getContext("2d");
+            ctx.reset();
+            ctx.fillStyle = "#ffffff";
+            ctx.textBaseline = 'middle';
+            var centerX = rootitem.width * 0.5;
+            var centerY = rootitem.height * 0.5;
+
+            var ourTime = wallClock.time.getHours();
+            var ourMins = wallClock.time.getMinutes();
+            var ourSecs = wallClock.time.getSeconds();
+            var ourDOW = Qt.formatDate(wallClock.time, "ddd").substr(0, 2);
+            var ourDay = Qt.formatDate(wallClock.time, "dd").substr(0, 2).replace(/^0/, '');
+
+            if (ourMins < 10) ourMins = "0" + ourMins;
+
+            function fontSizeCalc(ourSize) {
+                var ratio = ourSize / 280;
+                var size = rootitem.width * ratio;
+                return size|0;
+            }
+
+            if (use12H.value) {
+                if (ourTime >= 12) {
+                    ourTime -= 12;
+                    ctx.textAlign = "center";
+                    ctx.font = "bold " + fontSizeCalc(12) + "px " + "Roboto";
+					ctx.fillText("PM", centerX - (centerX * 0.45), centerY - (centerY * 0.09));
+                }
+
+                if (ourTime == 0) ourTime = 12;
+            } else {
+                ctx.textAlign = "center";
+                ctx.font = "bold " + fontSizeCalc(11) + "px " + "Roboto";
+                ctx.fillText("24H", centerX - (centerX * 0.29), centerY - (centerY * 0.085));
+            }
+
+            var fontFamily = "'Digital-7 Mono'";
+            ctx.textAlign = "right";
+            ctx.font = "50 " + fontSizeCalc(62) + "px " + fontFamily;
+            ctx.fillText(ourTime, centerX - (centerX * 0.143), centerY + (centerY * 0.05));
+
+            ctx.font = "50 " + fontSizeCalc(62) + "px " + fontFamily;
+            ctx.fillText(ourMins, centerX + (centerX * 0.32), centerY + (centerY * 0.05));
+			
+			ctx.font = "50 " + fontSizeCalc(30) + "px " + fontFamily;
+            ctx.fillText(ourDOW, centerX + (centerX * 0.05), centerY - (centerY * 0.22));
+
+            ctx.font = "50 " + fontSizeCalc(35) + "px " + fontFamily;
+            ctx.fillText(ourDay, centerX + (centerX * 0.61), centerY - (centerY * 0.21));
+
+            ctx.textAlign = "center";
+            ctx.font = "50 " + fontSizeCalc(46) + "px " + "Roboto";
+            ctx.fillText(":", centerX - (centerX * 0.103), centerY + (centerY * 0.15));
+        }
+        }
```

Comment:
This should be indented one tab left. l73, l122, l124 also have messed up indents

Reply:
I am in the process of fixing all the terrible indents.
I am trying to make the two canvases one, but the problem is that if i use something like
`            if (displayAmbient){
            ctx.fillStyle = "#ffffff";
            }else{
            ctx.fillStyle = "#252525";
            }`
the watchface will remain grey even after the screen turns off until it is next updated (which can take up to 20 seconds)

Reply:
it seems that the solution would be to rewrite the whole watchface to use qt text instead of qt canvas

Reply:
Could you try to use a `Connections` component https://doc.qt.io/qt-5/qml-qtqml-connections.html with `target` set to `desktop`, register an `onDisplayAmbientChanged` callback and call `requestPaint` on your `Canvas` https://doc.qt.io/qt-5/qml-qtquick-canvas.html#requestPaint-method ? I think this should solve your transition problem without needing a full rewrite nor code duplication

Reply:
I think that should all be done now